apps/services/user-profile/src/app/v2/test/user-profile.spec.ts

```diff
@@ -0,0 +1,132 @@
+import request from 'supertest'
+
+import { createCurrentUser } from '@island.is/testing/fixtures'
+import { UserProfileScope } from '@island.is/auth/scopes'
+import { setupApp, setupAppWithoutAuth } from '@island.is/testing/nest'
+
+import { AppModule } from '../../app.module'
+import { SequelizeConfigService } from '../../sequelizeConfig.service'
+import { FixtureFactory } from './fixtureFactory'
+
+const testNationalId = '0101302399'
+
+const testUserProfile = {
+  nationalId: testNationalId,
+  email: 'test@test.is',
+  mobilePhoneNumber: '1234567',
+}
+
+describe('UserProfileController', () => {
+  describe('No auth', () => {
+    it('GET /v2/users/.national-id should return 401 when user is not authenticated', async () => {
+      // Arrange
+      const app = await setupAppWithoutAuth({
+        AppModule: AppModule,
+        SequelizeConfigService: SequelizeConfigService,
+      })
+
+      const server = request(app.getHttpServer())
+
+      // Act
+      const res = await server
+        .get('/v2/users/.national-id')
+        .set('X-Param-National-Id', '1234567890')
+
+      // Assert
+      expect(res.status).toBe(401)
+      expect(res.body).toMatchObject({
+        statusCode: 401,
+        message: 'Unauthorized',
+      })
+
+      await app.cleanUp()
+    })
+
+    it('GET /v2/users/.national-id should return 403 Forbidden when user does not have the correct scope', async () => {
+      // Arrange
+      const app = await setupApp({
+        AppModule,
+        SequelizeConfigService,
+        user: createCurrentUser({}),
+      })
+
+      const server = request(app.getHttpServer())
+
+      // Act
+      const res = await server
+        .get('/v2/users/.national-id')
+        .set('X-Param-National-Id', '1234567890')
+
+      // Assert
+      expect(res.status).toEqual(403)
+      expect(res.body).toMatchObject({
+        statusCode: 403,
+        error: 'Forbidden',
+        message: 'Forbidden resource',
+      })
+
+      await app.cleanUp()
+    })
+  })
+
+  describe('With auth', () => {
+    let app = null
+    let server = null
+
+    beforeEach(async () => {
+      app = await setupApp({
+        AppModule,
+        SequelizeConfigService,
+        user: createCurrentUser({
+          scope: [UserProfileScope.system],
+        }),
+      })
+
+      server = request(app.getHttpServer())
+    })
+
+    afterEach(async () => {
+      await app.cleanUp()
+    })
```

Comment:
We usually try to set this up in a `beforeAll`/`afterAll` if it can be reused between tests, so the app is only initialised once. Then we can use `beforeEach`/`afterEach` to restore some state if neede, like truncating a db table or resetting a mock function calls.